src/ControlSystem/UpdateFunctionOfTime.hpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <typeinfo>
+#include <unordered_map>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp"
+
+namespace control_system {
+namespace Mutations {
+/// \ingroup ControlSystemGroup
+/// Updates a FunctionOfTime in the global cache. Intended to be used in
+/// Parallel::mutate.
+template <size_t DerivOrder>
+struct update_function_of_time {
+  static void apply(
+      const gsl::not_null<std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>*>
+          f_of_t_list,
+      const std::string& f_of_t_name, const double update_time,
+      const DataVector& update_deriv,
+      const double new_expiration_time) noexcept {
+    // First check if it's a PiecewisePolynomial
```

Comment:
Is the idea to add an entry here for every possible FOT that supports updating? That seems a bit hard to maintain, and doesn't play well with unit testing. If you have other plans for this, please let me know, else my suggestion would be to make `update` and `reset_expiration_time` virtual member function of the base class, perhaps with a default implementation that errors. @wthrowe thoughts?

Reply:
In theory we should only ever have to update PiecewisePolynomials and QuaternionFunctionsOfTime, but I see where you're coming from.

Reply:
I think the question here is whether we will add a FunctionOfTime that can update but doesn't use the arguments these two existing ones do.  I doubt we will, so having a virtual `update` seems fine to me.  The existing non-updating FoTs presumably all set their expiration time to infinity, so their implementations would be expected to error because you can't update before the expiration time.

Reply:
Ok I'll put in a separate PR for this and make this one dependent as I'll probably have to change things in different places.

Reply:
See #3499